test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageArchiveReaderTests.cs

```diff
@@ -1723,6 +1724,46 @@ public void GetContentHashForSignedPackage_WhenPackageNotSigned_ReturnsNull()
             }
         }
 
+#if IS_DESKTOP
+        [Fact]
+        public async Task GetContentHash_IsSameForUnsignedAndSignedPackageAsync()
+        {
+            // this test will create an unsiged package, copy it, then sign it. then compare the contentHash
+            var nupkg = new SimpleTestPackageContext("Package.Content.Hash.Test", "1.0.0");
+
+            using (var unsignedDir = TestDirectory.Create())
+            {
+                var nupkgFileName = $"{nupkg.Identity.Id}.{nupkg.Identity.Version}.nupkg";
+                var nupkgFileInfo = await nupkg.CreateAsFileAsync(unsignedDir, nupkgFileName);
+
+                using (var signedDir = TestDirectory.Create())
+                {
+                    Uri timestampService = null;
+                    var signatureHashAlgorithm = HashAlgorithmName.SHA256;
+                    var timestampHashAlgorithm = HashAlgorithmName.SHA256;
+
+                    var signedPackagePath = Path.Combine(signedDir.Path, nupkgFileName);
+
+                    using (var originalPackage = File.OpenRead(nupkgFileInfo.FullName))
+                    using (var signedPackage = File.Open(signedPackagePath, FileMode.OpenOrCreate, FileAccess.ReadWrite))
+                    using (var request = new AuthorSignPackageRequest(
+                        new X509Certificate2(),
```

Comment:
This is incorrect.  We use `X509Certificate2`'s default constructor in some argument validation tests, but it's not a valid signing certificate.  You should use `CertificatesFixture.GetDefaultCertificate()` like `AuthorSignPackageRequestTests` tests do.  Note that the certificate should be disposed too.